packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -589,16 +580,9 @@ class DeviceDomain extends Domain {
     registerHandler('forward', forward);
     registerHandler('unforward', unforward);
 
-    addDeviceDiscoverer(FuchsiaDevices());
-    addDeviceDiscoverer(AndroidDevices());
-    addDeviceDiscoverer(IOSDevices());
-    addDeviceDiscoverer(IOSSimulators());
-    addDeviceDiscoverer(FlutterTesterDevices());
-    if (flutterDesktopEnabled) {
-      addDeviceDiscoverer(MacOSDevices());
-      addDeviceDiscoverer(LinuxDevices());
-      addDeviceDiscoverer(WindowsDevices());
-    }
+    // Use the device manager discovery so that client provided device types
+    // are usable via the daemon protocol.
+    deviceManager.deviceDiscoverers.forEach(addDeviceDiscoverer);
```

Comment:
I don't believe that this change preserves the `if (flutterDesktopEnabled) {` check below.

Reply:
I think we could move that check into the deviceManager, since we've disabled those discoverers when the condition is false anyway

Reply:
Sounds good. I just want to make sure that we don't start reporting desktop devices until we know what the best way to present them in IDEs is.

Reply:
Sounds good.